src/models/uplift.py

```
@@ -0,0 +1,515 @@
+
+    qini = cum_treated_churn / t_count_safe - cum_control_churn / c_count_safe
+    qini *= np.arange(1, n + 1) / n   # scale by fraction of population
+
+    # Random baseline (diagonal)
+    frac = np.arange(1, n + 1) / n
+    baseline = frac * (
+        (y_true[treatment == 1].sum() / max(n_t, 1))
+        - (y_true[treatment == 0].sum() / max(n_c, 1))
+    )
+
+    _trapz = getattr(np, "trapezoid", None) or getattr(np, "trapz")
```

Ruff (0.15.12)

[warning] Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)